### PR TITLE
Add struct OperBag, turn macros into static inline, rename FunctionHeader -> FuncBag

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: src/opers.h [hunk 1]
```diff
@@ -27,78 +63,155 @@ extern Obj TRY_NEXT_METHOD;
...
 
 
 /****************************************************************************
 **
 *F  METHS_OPER( <oper> )  . . . . . . . . . . . . method list of an operation
 */
-#define METHS_OPER(oper,i)      (*            (ADDR_OBJ(oper) +21+ (i)) )
+static inline Obj METHS_OPER(Obj oper, Int i)
+{
+    return CONST_OPER(oper)->methods[i];
```
Comment (ChrisJefferson) [inline src/opers.h]: Could have bounds checks here and below in cache_oper

File: src/opers.h [hunk 2]
```diff
@@ -27,78 +63,155 @@ extern Obj TRY_NEXT_METHOD;
...
+
 
 /****************************************************************************
 **
 *F  ENABLED_ATTR( <oper> ) . . . . true if the operation is an attribute and
 **                                 storing is enabled (default) else false
 */
+static inline Int ENABLED_ATTR(Obj oper)
+{
+    Obj val = CONST_OPER(oper)->enabled;
```
Comment (ChrisJefferson) [inline src/opers.h]: Was trying to figure out why we go int -> obj -> int here, instead of just storing an int?

Comment (fingolfin) [reply src/opers.h]: The code previously did store 0 and 1 directly. But note that we use `MarkAllSubBags` for `T_FUNCTION`, so strictly speaking, any any data in there should be a valid bag / object. Now, by coincidence, the integers `0` and `1` are fine (the latter being the immediate integer encoding for 0). But I felt it was wrong to do that blindly. 

That said, I could restore the old behavior, and add a comment which points out what is going on; I'd also enforce that only 0 and 1 get stored. (BTW, it's a bit annoying that we waste a full word to essentially store a single bit).

File: src/opers.h (hunk 1 above)
Comment (fingolfin) [reply src/opers.h]: Added the bounds checks.

File: src/opers.h (hunk 2 above)
Comment (ChrisJefferson) [reply src/opers.h]: The fact we call markAllSubbags is good enough, and I definately wouldn't want to start relying on how we happen to store immediate integers :)

**Comment by fingolfin** [comment]
See also issue #1615: By having this as a struct, we could now think about adding a few extra bits into the space currently occupied by the `ENABLED_ATTR` bit along. E.g. an `IS_FILTER` bit to explicitly mark filters as such.

**Comment by fingolfin** [comment]
@ChrisJefferson any further thoughts?

**Pull request merged**